jerry-core/jerry.c

```diff
@@ -2121,6 +2162,85 @@ jerry_foreach_object_property (const jerry_value_t obj_val, /**< object value */
 } /* jerry_foreach_object_property */
 
 /**
+ * Common part for jerry_resolve_promise and jerry_reject_promise.
+ *
+ * @return undefined value - if success
+ *         value marked with error flag - otherwise
+ */
+static jerry_value_t
+jerry_resolve_or_reject_promise (jerry_value_t promise,
+                                 jerry_value_t argument,
+                                 bool is_resolve)
```

Comment:
I think it makes sense to expose something like this in the public API.

I think people will be writing code like in [this example] a lot, so you naturally have a `bool` to make the decision whether `resolve` or `reject` should be called.

In an [earlier sketch], I used an `enum`, I think that has a bit better readability vs `bool`, but I'm not feeling strongly about it.

[this example]: https://github.com/jerryscript-project/jerryscript/issues/1794#issuecomment-298308945
[earlier sketch]: https://github.com/jerryscript-project/jerryscript/issues/1794#issuecomment-298319724